testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -682,7 +606,8 @@ impl BlockMinerThread {
         )
         .expect("FATAL: could not open sortition DB");
 
-        Self::broadcast_p2p(&sort_db, &mut chain_state, &block, reward_set)
+        // push block via p2p block push
+        self.broadcast_p2p(&sort_db, &mut chain_state, &block, reward_set)
             .map_err(NakamotoNodeError::AcceptFailure)?;
 
         let Some(ref miner_privkey) = self.config.miner.mining_key else {
```

Comment:
This wasn't a change of yours, but should a block be broadcasted if there is no mining key ? I almost expect this check to occur before the broadcast_p2p call.

Reply:
Sure